packages/connector-core/src/index.ts

```diff
@@ -0,0 +1,30 @@
+import { ZodType } from 'zod';
+
+import { ConnectorError, ConnectorErrorCodes, GetConnectorConfig } from './types';
+
+export * from './types';
+
+// eslint-disable-next-line @silverhand/fp/no-let
+let _getConfig: GetConnectorConfig = async () => {
```

Comment:
maybe we can create a default options object and extend it to prevent mutating like ky?

Reply:
Updated with new design: createConnector function, PTAL